backend/services/theme_service.py

```
@@ -77,97 +77,343 @@ def _create_smart_person_background_mask(self, image_bytes: bytes, image_width:
+            original_image = Image.open(io.BytesIO(image_bytes)).convert('RGB')
+            original_array = np.array(original_image)
+            
+            # Validate image dimensions match parameters
+            actual_height, actual_width = original_array.shape[:2]
+            if actual_width != image_width or actual_height != image_height:
+                logger.warning(f"Image dimensions mismatch: expected {image_width}x{image_height}, got {actual_width}x{actual_height}")
+                # Use actual dimensions from image
+                image_width, image_height = actual_width, actual_height
+                
+        except Exception as e:
+            raise RuntimeError(f"Failed to load or process image: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)